src/Shared/Recruit.Vacancies.Client/Application/CommandHandlers/ApproveReferredVacancyReviewCommandHandler.cs

```diff
@@ -3,46 +3,57 @@
 using MediatR;
 using System.Threading;
 using System.Threading.Tasks;
-using System;
 using Esfa.Recruit.Vacancies.Client.Domain.Entities;
 using Esfa.Recruit.Vacancies.Client.Domain.Messaging;
 using Esfa.Recruit.Vacancies.Client.Domain.Events;
-using Esfa.Recruit.Vacancies.Client.Domain.Exceptions;
+using Microsoft.Extensions.Logging;
 
 namespace Esfa.Recruit.Vacancies.Client.Application.CommandHandlers
 {
     public class ApproveReferredVacancyReviewCommandHandler: IRequestHandler<ApproveReferredVacancyReviewCommand>
     {
+        private readonly ILogger<ApproveReferredVacancyReviewCommandHandler> _logger;
         private readonly IVacancyReviewRepository _vacancyReviewRepository;
         private readonly IVacancyRepository _vacancyRepository;
         private readonly IMessaging _messaging;
 
-        public ApproveReferredVacancyReviewCommandHandler(IVacancyReviewRepository vacancyReviewRepository, IVacancyRepository vacancyRespository, IMessaging messaging)
+        public ApproveReferredVacancyReviewCommandHandler(
+                            ILogger<ApproveReferredVacancyReviewCommandHandler> logger, 
+                            IVacancyReviewRepository vacancyReviewRepository, 
+                            IVacancyRepository vacancyRespository, 
+                            IMessaging messaging)
         {
+            _logger = logger;
             _vacancyReviewRepository = vacancyReviewRepository;
             _vacancyRepository = vacancyRespository;
             _messaging = messaging;
         }
 
         public async Task Handle(ApproveReferredVacancyReviewCommand message, CancellationToken cancellationToken)
         {
+            _logger.LogInformation("Approving referred review {reviewId} for vacancy {vacancyReference}.", message.ReviewId, message.Vacancy.VacancyReference);
+            
             var vacancy = await _vacancyRepository.GetVacancyAsync(message.Vacancy.Id);
```

Comment:
I think we should change the ApproveReferredVacancyReviewCommand so it only takes a VacancyId and not a full Vacancy.

Reply:
Where would you do the update of the vacancy?